src/ReinforcementLearningCore/src/utils/networks.jl

```diff
@@ -255,8 +255,9 @@ Transform a vector containing the non-zero elements of a lower triangular da x d
 function vec_to_tril(cholesky_vec, da)
```

Comment:
JET finds lots of type issues with this function. Based on a scaffolding from ChatGPT, I came up with:

```julia
function vec_to_tril(v::AbstractVector{T}) where {T<:AbstractFloat}
    n = length(v)
    m = Int(floor(sqrt(2 * n + 1)))
    matrix = Matrix{T}(undef, m, m)

    # Fill the lower triangular part of the matrix with the elements of the vector
    index = 1
    for j = 1:m
        for i = j:m
            matrix[i, j] = i == j ? softplusbeta(v[index]) : v[index]
            index += 1
        end
    end

    return matrix
end
```

Reply:
this is significantly faster and fewer allocations, but likely there are other (GPU???) downsides which are why this wasn't implemented originally. One difference (which could be fixed is that this function returns a matrix instead of a 3 dimensional array where the third dimension has a length of one.

Reply:
Sadly this implementation is both Zygote incompatible due to array mutation and GPU incompatible due to scalar indexing. Coming up with a function that satisfies both of these requirements is the reason behind the weird design of the function in its current state. I think the Enzyme AD will solve the mutation problem and then it might be possible to also make a better GPU kernel.

Reply:
Figures. What do you think about pulling out the three functions that are defined within `vec_to_tril` so it's easier to read? It'd also be easier to test them, though I don't think that's strictly necessary.